support/oidc-discovery-provider/handler.go

```diff
@@ -34,14 +44,31 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if _, ok := h.domains[r.Host]; !ok {
+		http.Error(w, "domain not allowed", http.StatusNotFound)
+		return
+	}
+
+	if h.allowInsecureScheme {
+		if r.URL.Scheme == "" {
+			if r.TLS == nil {
+				r.URL.Scheme = "http"
+			} else {
+				r.URL.Scheme = "https"
+			}
+		}
+	} else {
+		r.URL.Scheme = "https"
+	}
```

Comment:
nit: can be simplified for readability:
```suggestion
	urlScheme := "https"
	if h.allowInsecureScheme && r.URL.Scheme == "" && r.TLS == nil {
		urlScheme = "http"
	}
```
Followed by the use of `urlScheme` when constructing the URLs below.